static/src/js/components/CollectionSelector/__tests__/CollectionSelector.test.jsx

```diff
@@ -0,0 +1,325 @@
+                      s3BucketAndObjectPrefixNames: [
+                        's3://example1',
+                        's3://example2',
+                        'prefix_bucket/*'
+                      ],
+                      s3CredentialsApiEndpoint: 'https://example.org',
+                      s3CredentialsApiDocumentationUrl: 'https://example.org'
+                    },
+                    shortName: 'Collection 1',
+                    provider: 'MMT_2',
+                    entryTitle: 'Mock title of collection 1',
+                    __typename: 'Collection'
+                  },
+                  {
+                    conceptId: 'C1200482349-MMT_2',
+                    directDistributionInformation: null,
+                    shortName: 'Collection 2',
+                    provider: 'MMT_2',
+                    entryTitle: 'Mock title of collection 2',
+                    __typename: 'Collection'
+                  }
+                ],
+                __typename: 'CollectionList'
+              }
+            }
+          }
+        }, ...additionalMocks]
+      }
+      >
+        <Suspense>
+          <CollectionSelector {...props} />
+        </Suspense>
+      </MockedProvider>
+    </AuthContext.Provider>
+  )
+
+  return {
+    props,
+    user
+  }
+}
+
+describe('CollectionSelector', () => {
+  describe('when adding item to selected collection', () => {
+    test('should add the selected collection and calls onChange', async () => {
+      const { user, props } = setup({})
+
+      const nameField = await screen.findByText('Collection 1 | Mock title of collection 1')
+      await user.click(nameField)
+
+      const addIcon = screen.getByRole('button', { name: '+ icon' })
+      await user.click(addIcon)
+
+      expect(screen.getByText('Showing selected 1 items')).toBeInTheDocument()
+
+      expect(props.onChange).toHaveBeenCalledTimes(1)
+      expect(props.onChange).toHaveBeenCalledWith([
+        {
+          __typename: 'Collection',
+          conceptId: 'C1200444618-MMT_2',
+          directDistributionInformation: {
+            region: 'us-east-2',
+            s3BucketAndObjectPrefixNames: [
+              's3://example1',
+              's3://example2',
+              'prefix_bucket/*'
+            ],
+            s3CredentialsApiDocumentationUrl: 'https://example.org',
+            s3CredentialsApiEndpoint: 'https://example.org'
+          },
+          entryTitle: 'Mock title of collection 1',
+          provider: 'MMT_2',
+          shortName: 'Collection 1'
+        }
+      ])
+    })
+  })
+
+  describe('when removing a selected item', () => {
+    test('should move the item back to available and calls onChange', async () => {
+      const { user, props } = setup({})
+
+      const nameField = await screen.findByText('Collection 1 | Mock title of collection 1')
+
+      // Tests selecting and unselecting the available option
+      await user.click(nameField)
```

Comment:
I'm confused why there are 3 clicks here, selecting and unselecting is only two actions. 

It is a good idea to perform an assertion on something after user actions, especially if the user action is the same each time